validationparser/interceptor.go

```diff
@@ -0,0 +1,46 @@
+package validationparser
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type validator interface {
+	Validate() error
+}
+
+// UnaryServerInterceptor returns grpc.UnaryServerInterceptor
+// that should be used as a middleware that parser through the protoc-gen-validate middleware and returns a
+// user friendly error.
+func UnaryServerInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		if v, ok := req.(validator); ok {
+			if err := v.Validate(); err != nil {
+				// Check what type of error we received
+				typeOfError := reflect.TypeOf(err)
+				valueOfError := reflect.ValueOf(err)
+				// If the error is a type of validation error then parse
+				if strings.Contains(typeOfError.String(), "ValidationError") && valueOfError.NumField() == 4 {
```

Comment:
this is a pretty tenuous assumption... Are you saying that it's a `lyft/validate` error if it ends with "ValidationError"?

Reply:
it's also kind of weird to own middleware that affects a specific open source project without it being in the open source project. We're basically "hacking" this project instead of adding value to the project we're relying on. Does that make any sense?